plottr/node/scaleunits.py

```diff
@@ -0,0 +1,136 @@
+    def unitscale_button_selected(
+            self,
+            btn: QtWidgets.QAbstractButton,
+            checked: bool
+    ) -> None:
+        if checked:
+            self.unit_scale_selected.emit(ScaleUnitsOption(self.btnGroup.id(btn)))
+
+
+class ScaleUnitsWidget(NodeWidget):
+    """Node widget for :class:`ScaleUnits`."""
+
+    def __init__(self, node: Optional[Node] = None):
+        super().__init__(node=node, embedWidgetClass=ScaleUnitOptionWidget)
+        assert self.widget is not None
+        self.widget.unit_scale_selected.connect(
+            lambda x: self.signalOption('scale_unit_option')
+        )
+
+        self.optSetters = {
+            'scale_unit_option': self.set_scale_unit_option,
+        }
+
+        self.optGetters = {
+            'scale_unit_option': self.get_scale_units_option,
+        }
+
+    def get_scale_units_option(self) -> ScaleUnitsOption:
+        assert self.widget is not None
+        activeBtn = self.widget.btnGroup.checkedButton()
+        activeId = self.widget.btnGroup.id(activeBtn)
+        return ScaleUnitsOption(activeId)
+
+    def set_scale_unit_option(self, option: ScaleUnitsOption) -> None:
+        assert self.widget is not None
+        self._emitUpdate = False
+        for k, btn in self.widget.buttons.items():
+            if k == option:
+                btn.setChecked(True)
+        self._emitUpdate = True
+
+
+class ScaleUnits(Node):
+    """
+    A Node that automatically scales the units and values such that
+    for example 1e-9 V will be rendered as 1 nV. The logic for how to scale units
+    in inherited from QCoDeS. Basically pure SI units are scaled with enginering prefixes
+    while more complex units are scaled by adding a power of 10 prefix to the unit
+    e.g (1*10**3 complexunit)
+    """
+    useUi = True
+    uiClass = ScaleUnitsWidget
+
+    def __init__(self, name: str):
+        super().__init__(name)
+        self._scale_unit_option: ScaleUnitsOption = ScaleUnitsOption.always
+
+    @property
+    def scale_unit_option(self) -> ScaleUnitsOption:
+        return self._scale_unit_option
+
+    @scale_unit_option.setter  # type: ignore[misc]
+    @updateOption('scale_unit_option')
+    def scale_unit_option(self, value: ScaleUnitsOption) -> None:
+        self._scale_unit_option = value
+
+    def process(self, dataIn: Optional[DataDictBase] = None) -> Optional[Dict[str, Optional[DataDictBase]]]:
+        if super().process(dataIn=dataIn) is None:
+            return None
+        assert dataIn is not None
+        data = dataIn.copy()
+
+        for name, data_item in data.data_items():
+            if self.scale_unit_option != ScaleUnitsOption.never:
+                prefix, selected_scale = find_scale_and_prefix(
+                    data_item['values'],
+                    data_item["unit"]
+                )
+                data_item["unit"] = prefix + data_item["unit"]
+                data_item['values'] = data_item['values'] * 10**(-selected_scale)
```

Comment:
i remember there was a comment about categorical data - is it out of scope or did it get resloved? if i remember correctly, in qcodes, categorical data does not get unit-scaled, so perhaps we should do the same here?

Reply:
Plottr does not in general support categorical data